manager/integration/tests/test_cloning.py

```diff
@@ -244,29 +244,32 @@ def test_cloning_with_detached_source_volume(client, core_api, pvc, clone_pvc):
     source_volume = wait_for_volume_attached(client, source_volume_name)
 
     # Step-7
-    wait_for_snapshot_count(source_volume, 2)
-
-    # Step-8
     clone_volume_name = get_volume_name(core_api, clone_pvc_name)
     wait_for_volume_clone_status(client, clone_volume_name, VOLUME_FIELD_STATE,
                                  VOLUME_FIELD_CLONE_COMPLETED)
     wait_for_volume_detached(client, clone_volume_name)
 
-    # Step-9
+    # Step-8
     wait_for_volume_detached(client, source_volume_name)
 
-    # Step-10
+    # Step-9
     clone_volume = client.by_id_volume(clone_volume_name)
     clone_volume.attach(hostId=lht_host_id)
     wait_for_volume_attached(client, clone_volume_name)
     clone_volume = wait_for_volume_endpoint(client, clone_volume_name)
 
-    # Step-11
+    # Step-10
     check_volume_data(clone_volume, data)
 
-    # Step-12
+    # Step-11
     wait_for_volume_healthy(client, clone_volume_name)
 
+    # Step-12
+    source_volume = client.by_id_volume(source_volume_name)
+    source_volume.attach(hostId=lht_host_id)
+    source_volume = wait_for_volume_healthy(client, source_volume_name)
```

Comment:
I think the appropriate check would be to wait for volume to get attached instead of healthy.

Reply:
Can you elaborate more about the difference?

Reply:
`wait_for_volume_healthy` has more checks than `wait_for_volume_attached`. It will first check whether volume is attached and then checks about the robustness of the volume. So if you just intend to check the snapshot count, after the volume is attached, you should be able to get that. I don't think the intention here to check the robustness.

Reply:
Thank you for the explanation! Refined!